catboost/private/libs/algo/greedy_tensor_search.cpp

```diff
@@ -760,8 +759,7 @@ static void CalcBestScoreLeafwise(
     double scoreStDev,
     TVector<TCandidatesContext>* candidatesContexts, // [dataset]
     TFold* fold,
-    TLearnContext* ctx) {
-
+    TLearnContext* ctx, TBucketStats* pstats = nullptr, TBucketStats* parent_pstats = nullptr, TBucketStats* sibling_pstats = nullptr, int maxBucketCount = 0, int maxSplitEnsamples = 0) {
```

Comment:
please avoid abbreviations like pstats, sstats, etc. -- we use meaningful names

please use TArrayRef and TConstArrayRef instead of pointers

Reply:
done